src/ReorderLiteralsTransformer.cpp

```diff
@@ -331,7 +331,7 @@ std::vector<unsigned int> applySips(sips_t sipsFunction, std::vector<AstAtom*> a
 
 bool reorderClauseWithSips(sips_t sipsFunction, AstClause* clause) {
     // ignore clauses with fixed execution plans
-    if (clause->hasFixedExecutionPlan()) {
+    if (clause->getExecutionPlan() != nullptr) {
```

Comment:
Could the execution plan ever be modified elsewhere? i.e in which case you should also check hasFixed. (and maybe set it when you modify the order?)